modules/junit-jupiter/src/main/java/org/testcontainers/junit/jupiter/TestcontainersExtension.java

```diff
@@ -83,7 +84,13 @@ public void beforeEach(final ExtensionContext context) {
     }
 
     private static Predicate<Field> isContainer() {
-        return field -> Startable.class.isAssignableFrom(field.getType()) && AnnotationSupport.isAnnotated(field, Container.class);
+        return field -> {
+            boolean isStartable = Startable.class.isAssignableFrom(field.getType());
+            if (!isStartable) {
+                throw new ExtensionConfigurationException("Annotation is only supported for Startable types");
```

Comment:
I think we need to first check that it is annotated with `Container` and only then fail if it is not startable

Reply:
lol, yes

Reply:
And the CI correctly failed :D